worlds/pokemon_crystal/options.py

```diff
@@ -0,0 +1,374 @@
+from dataclasses import dataclass
+
+from Options import Toggle, Choice, DefaultOnToggle, Range, PerGameCommonOptions, NamedRange
+
+
+class Goal(Choice):
+    """Elite Four: collect 8 badges and enter the Hall of Fame
+        Red: collect 16 badges and defeat Red at Mt. Silver"""
+    display_name = "Goal"
+    default = 0
+    option_elite_four = 0
+    option_red = 1
+
+
+class JohtoOnly(Choice):
+    """Excludes all of Kanto, disables early Kanto access
+        Forces Goal to Elite Four unless Silver Cave is included.
+        Goal badges will be limited to 8 if badges are shuffled or vanilla"""
+    display_name = "Johto Only"
+    default = 0
+    option_off = 0
+    option_on = 1
+    option_include_silver_cave = 2
+
+
+class EliteFourBadges(Range):
+    """Number of badges required to enter Victory Road"""
+    display_name = "Elite Four Badges"
+    default = 8
+    range_start = 1
+    range_end = 16
+
+
+class RedBadges(Range):
+    """Number of badges required to open Silver Cave"""
+    display_name = "Red Badges"
+    default = 16
+    range_start = 1
+    range_end = 16
+
+
+class RandomizeBadges(Choice):
+    """Shuffles badges into the pool"""
+    display_name = "Randomize Badges"
+    default = 2
+    option_vanilla = 0
+    option_shuffle = 1
+    option_completely_random = 2
+
+
+class RandomizeHiddenItems(Toggle):
+    """Shuffles hidden items into the pool"""
+    display_name = "Randomize Hidden Items"
+    default = 0
+
+
+class RequireItemfinder(DefaultOnToggle):
+    """Hidden items require Itemfinder in logic"""
+    display_name = "Require Itemfinder"
+
+
+class Trainersanity(Toggle):
+    """Adds checks for defeating trainers"""
+    display_name = "Trainersanity"
+    default = 0
+
+
+class TrainersanityAlerts(Choice):
+    """Shows a message box or plays a sound for trainersanity checks"""
+    display_name = "Trainersanity"
+    default = 1
+    option_no_alerts = 0
+    option_message_box = 1
+    option_sound_only = 2
+
+
+class RandomizePokegear(Toggle):
+    """Shuffles the Pokegear and cards into the pool"""
+    display_name = "Randomize Pokegear"
+    default = 0
+
+
+class RandomizeBerryTrees(Toggle):
+    """Shuffles berry tree items into the pool"""
```

Comment:
I believe this is referring to locations and not items
```suggestion
    """Shuffles berry tree locations into the pool"""
```